---
title: "Lef1_G4P_bedtoolAnalysis"
author: "Abigail E Hein"
date: "2/8/2022"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Dataset Download
Download Lef1 CHIP-seq data from GSM2825571
  File name: GSM2825571_ENCFF072ERX_peaks_hg19.bed.gz
Download G4P-CHIP-seq data from GSE133379
  File name: GSE133379_293T-G4P-hg19-rep1.narrowPeak.gz

  
genome build: hg19 
  File name: /opt/bedtools/2.23.0/genomes/human.hg19.genome
  hg19 fasta file name: hg19.fa 
  

```{bash}
cd ~/GQ_HMGB_CHIPanalysis/Lef1

wget -q https://ftp.ncbi.nlm.nih.gov/geo/samples/GSM2825nnn/GSM2825571/suppl/GSM2825571_ENCFF072ERX_peaks_hg19.bed.gz

wget -q https://ftp.ncbi.nlm.nih.gov/geo/series/GSE133nnn/GSE133379/suppl/GSE133379_293T-G4P-hg19-rep1.narrowPeak.gz

md5sum *.gz > md5sums.txt

gunzip *.gz
```

## Intersect Datasets

```{bash engine.opts='-l'}
wc -l GSM2825571_ENCFF072ERX_peaks_hg19.bed

wc -l GSE133379_293T-G4P-hg19-rep1.narrowPeak

bedtools intersect -a GSM2825571_ENCFF072ERX_peaks_hg19.bed -b GSE133379_293T-G4P-hg19-rep1.narrowPeak > Int_Lef1_G4P.bed

wc -l Int_Lef1_G4P.bed
```

## Format BED files for further analysis
BED files sorted by chromosome
  File name for sorted Lef1 CHIP-seq: sorted_lef1_293T_chipSeq.bed
  File name for sorted G4P-CHIP-seq: sorted_G4P_293T_chipSeq.bed
  File name for sorted Lef1-G4P intersect: sorted_int_lef1-G4P_293T.bed
  File name for sorted hg19 intersect: sorted_human.hg19.genome
  
Minimized Bed- columns 1-6
  File name for minimized Lef1 CHIP-seq: col1-6_sorted_lef1_293T_chipSeq.bed
  File name for minimized G4P-CHIP-seq: col1-6_sorted_G4P_293T_chipSeq.bed
  File name for minimized Lef1-G4P intersect: col1-6_sorted_int_lef1-G4P_293T.bed

```{bash engine.opts='-l'}
sort -k1,1 -k2,2n GSM2825571_ENCFF072ERX_peaks_hg19.bed > sorted_lef1_293T_chipSeq.bed  

sort -k1,1 -k2,2n GSE133379_293T-G4P-hg19-rep1.narrowPeak > sorted_G4P_293T_chipSeq.narrowPeak  

sort -k1,1 -k2,2n Int_Lef1_G4P.bed > sorted_int_lef1-G4P_293T.bed  

sort -k1,1 -k2,2n /opt/bedtools/2.23.0/genomes/human.hg19.genome > sorted_human.hg19.genome 

cut -f 1-6 sorted_lef1_293T_chipSeq.bed > col1-6_sorted_lef1_293T_chipSeq.bed

cut -f 1-6 sorted_G4P_293T_chipSeq.narrowPeak > col1-6_sorted_G4P_293T_chipSeq.bed

cut -f 1-6 sorted_int_lef1-G4P_293T.bed > col1-6_sorted_int_lef1-G4P_293T.bed

```

## Convert BED files to fasta
reference genome fasta file name: hg19.fa
  md5sum= 806c02398f5ac5da8ffd6da2d1d5d1a9  hg19.fa.gz

```{bash engine.opts='-l'}
wget -q https://hgdownload.soe.ucsc.edu/goldenPath/hg19/bigZips/hg19.fa.gz

echo hg19_md5sum
md5sum hg19.fa.gz

gunzip hg19.fa.gz
```

Generated Fasta Files
  File name for sorted Lef1 CHIP-seq: lef1_293T_chipSeq.fa
  File name for sorted G4P-CHIP-seq: G4P_293T_chipSeq.fa
  File name for sorted Lef1-G4P intersect: Int_lef1_G4P_293T.fa
```{bash engine.opts='-l'}
bedtools getfasta -fi hg19.fa -bed GSM2825571_ENCFF072ERX_peaks_hg19.bed -fo lef1_293T_chipSeq.fa

bedtools getfasta -fi hg19.fa -bed GSE133379_293T-G4P-hg19-rep1.narrowPeak -fo G4P_293T_chipSeq.fa

bedtools getfasta -fi hg19.fa -bed Int_Lef1_G4P.bed -fo Int_lef1_G4P_293T.fa
```


