---
title: "sox2_GQ_fuctionalSites"
author: "Abigail E Hein"
date: "2/11/2022"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(pqsfinder)
library(Biostrings)
library(ggplot2)
library(tidyr)
library(RColorBrewer)
library(ChIPpeakAnno)
library(regioneR)
library(TxDb.Hsapiens.UCSC.hg19.knownGene)
library(ChIPseeker)
library(ReactomePA)
library(org.Hs.eg.db)
library(dplyr)
library(annotate)
library(DOSE)
```



```{r}
sox2Sites <- toGRanges("col1-6_sorted_sox2_H9_chipSeq.bed")
G4Sites <- toGRanges("col1-6_sorted_G4P_H9_chipSeq.bed")
IntersectedSites <- toGRanges("col1-6_sorted_int_sox2-G4P_H9.bed")
IntersectedSitesSeq <- readDNAStringSet("Int_sox2_G4P_H9.fa", format = "fasta")
```


## R Markdown

This is an R Markdown document. Markdown is a simple formatting syntax for authoring HTML, PDF, and MS Word documents. For more details on using R Markdown see <http://rmarkdown.rstudio.com>.

When you click the **Knit** button a document will be generated that includes both content as well as the output of any embedded R code chunks within the document. You can embed an R code chunk like this:

```{r}

#pqsFinder on enhancer regions
# load gene annotation from UCSC for human genome build hg19
txdb <- TxDb.Hsapiens.UCSC.hg19.knownGene #shortcut

# get all genes from the data base as GRanges object
genes <- genes(txdb)

# take the the region around the gene start as promoter
prom <- promoters(genes, upstream=1000, downstream=100)

#annotate sox2-G4 peaks
sox2Anno <- annotatePeak(sox2Sites, TxDb=TxDb.Hsapiens.UCSC.hg19.knownGene)
plotAnnoPie(sox2Anno)
sox2G4anno <- annotatePeak(IntersectedSites, TxDb=TxDb.Hsapiens.UCSC.hg19.knownGene)
plotAnnoPie(sox2G4anno)
G4anno <- annotatePeak(G4Sites, TxDb=TxDb.Hsapiens.UCSC.hg19.knownGene)
plotAnnoPie(G4anno)
```

## Pathway entichment
```{r}
#enrichment analysis
enrichmentResults <- enrichPathway(sox2G4anno@anno$geneId)
dotplot(enrichmentResults)
emapplot(enrichmentResults, color = "pvalue")
erich <- setReadable(enrichmentResults, 'org.Hs.eg.db', 'ENTREZID')
cnetplot(erich, node_label="category", cex_label_category = 1.2)
enrichmentLeef <- enrichPathway(sox2Anno@anno$geneId)
dotplot(enrichmentLeef)
erichL <- setReadable(enrichmentLeef, 'org.Hs.eg.db', 'ENTREZID')
cnetplot(erichL, node_label="category", cex_label_category = 1.2)
#emapplot(enrichmentLeef, color = "pvalue")
enrichmentG4 <- enrichPathway(G4anno@anno$geneId)
dotplot(enrichmentG4)
emapplot(enrichmentG4, color = "pvalue")
erichG <- setReadable(enrichmentG4, 'org.Hs.eg.db', 'ENTREZID')
cnetplot(erichG, node_label="category", cex_label_category = 1.2)
```

## sox2-G4 promoter distributions
  We search for overlap between ER peaks and promoters

```{r}
sox2G4_prom_overlap_subset <- subset(sox2G4anno, sox2G4anno@anno$annotation== "Promoter (<=1kb)")
write.table(sox2G4_prom_overlap_subset, "sox2G4_prom_overlap.bed", quote = F, row.names = F, col.names = F, sep="\t")
```


## Convert promoter hits to fasta w/ bedtools


```{bash engine.opts='-l'}
bedtools getfasta -fi hg38.fa -bed sox2G4_prom_overlap.bed -fo sox2GQ_H1_prom_int.fa
cut -f 1-6 sox2G4_prom_overlap.bed > col1-6_sox2G4_prom_overlap.bed
```



```{r include=FALSE}
sox2G4_prom_seq <- readDNAStringSet("sox2GQ_H1_prom_int.fa", format = "fasta")
sox2G4_prom_overlap_subset_bed <- read.table("sox2G4_prom_overlap.bed", sep = "\t", stringsAsFactors = F)
colnames(sox2G4_prom_overlap_subset_bed) <- c("peakChr", "peakStart", "peakEnd", "peakLength",  "peakStrand", "annotation", "geneChr", "geneStart", "geneEnd", "geneLength", "geneStrand", "geneId", "transcriptId", "distanceToTSS")
rm <- vector()
for(ii in 1:nrow(sox2G4_prom_overlap_subset_bed)) {
  if(sox2G4_prom_overlap_subset_bed[ii,4] == 1){
    rm <- append(rm, ii)
  }
}
sox2G4_prom_overlap_subset_bed <- sox2G4_prom_overlap_subset_bed[-c(rm),]


#pqsFinder on promoter regions
winnersI <- vector()
numWinnersI <- 0
numWinnerSites <- 0
winnersIscore <-data.frame(score= integer(), OG_file_row=numeric())

#NEED TO DO THIS AT MIN=25 JUST FOR COUNT
for(ii in 1:length(sox2G4_prom_seq)) {
  pqs <- pqsfinder(sox2G4_prom_seq[[ii]], min_score = 52)
  dss <- as(pqs, "DNAStringSet")
  if (length(pqs@elementMetadata@listData[["score"]]) > 0) {
    numWinnerSites <- numWinnerSites +1
    numWinnersI <- numWinnersI + length(pqs@elementMetadata@listData[["score"]])
    winnersI[numWinnerSites] <- ii
    hit <- data.frame(pqs@elementMetadata@listData[["score"]], rep(ii, length(pqs@elementMetadata@listData[["score"]])))
    winnersIscore <-  rbind(winnersIscore, hit)
  }
  writeXStringSet(dss, file = "pred_GQ_prom.txt", format = "fasta", append = T)
}

colnames(winnersIscore) <- c("score", "og_df_row")
winnersIscore <- cbind(winnersIscore, seq(1:length(winnersIscore$score)))
colnames(winnersIscore) <- c("score", "og_df_row", "winnerNo")
winnersIscore <- winnersIscore[order(winnersIscore$winnerNo),]



prom_pqs_sox2G4_hits <- data.frame(peakChr = character(), peakStart = integer(), peakEnd = integer(), peakLength = integer(), peakStrand = integer(), name = logical(), score = integer(), annotation = character(), geneChr = integer(), geneStart = integer(), geneEnd = integer(), geneLength = integer(), geneStrand = integer(), geneId = integer(), transcriptId = character(), distanceToTSS  = integer(), stringsAsFactors = F)
for(ii in 1:nrow(winnersIscore)) {
  prom_pqs_sox2G4_hits[ii,] <- sox2G4_prom_overlap_subset_bed[winnersIscore[ii,2],]
}
write.csv(prom_pqs_sox2G4_hits, "G4prom_sox2.txt", row.names = F, quote = F)


# take only unique ids
gene.ids <- unique(prom_pqs_sox2G4_hits$geneId)
gene.symb <- getSYMBOL(as.character(gene.ids), data='org.Hs.eg')
gene.ids <-cbind(gene.ids, gene.symb)

# write names to an output file
write.csv(gene.ids, file="sox2G4_regulated_genes.txt", quote=FALSE, row.names=FALSE)

write.csv(winnersIscore, "sox2GQ_promoters.csv", row.names = F)
```


## Including Plots
```{bash}
awk 'NR%2{printf "%s ",$0;next;}1' pred_GQ_prom.txt > pred_GQ_prom_compressed.txt
```


```{r}
sox2G4_minimized_hits <- read.table("pred_GQ_prom_compressed.txt", sep =';')
sox2G4_minimized_hits <- cbind(sox2G4_minimized_hits[,3:16], prom_pqs_sox2G4_hits)
write.csv(sox2G4_minimized_hits, "G4prom_sox2.csv", row.names = F, quote = F)
```

```{bash engine.opts='-l'}
wget -q http://www.enhanceratlas.org/data/download/enhancer/hs/H9.bed

bedtools intersect -a Int_Sox2_G4.bed -b  H9_enhancers_Hg38.bed > sox2_G4_enhancers.bed

sort -k1,1 -k2,2n H9_enhancers_Hg38.bed > sorted_H9_enhancers_Hg38.bed

bedtools fisher -a sorted_int_sox2-G4P_H9.bed -b sorted_H9_enhancers_Hg38.bed -g sorted_human.hg38.genome
```

